### PR TITLE
Pin the nightly version in CI

### PR DESCRIPTION
**Pull request by gregtatum**
This is part of the work for #374. I have several PRs with red CI. I'm keeping this as draft while I work on it.

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -268,8 +268,10 @@ jobs:
 
       # Nightly is only needed for the `--profile bench` line. Once that is stabilized,
       # then this line is no longer needed.
-      - name: Install nightly toolchain
-        run: rustup toolchain install nightly
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly-2020-04-04
```
Comment (sffc) [inline .github/workflows/build-test.yml:273]: Why are you picking such old versions of nightly?  Seems like you should pick the most recent version that works (something in 2021, not 2020)

**Comment by sffc** [comment]
Closer!  Coverage is working now, but Memory is still broken:

```
     Running `target/debug/icu_benchmark_memory --os ubuntu-latest icu_datetime/work_log icu_locid/syntatically_canonicalize_locales icu_locid/filter_langids icu_plurals/unread_emails icu_plurals/elevator_floors icu_uniset/unicode_bmp_blocks_selector fixed_decimal/permyriad writeable/writeable_message`
[memory] Starting example "work_log"
[memory] > error: toolchain 'nightly-x86_64-unknown-linux-gnu' is not installed
The example "work_log" had a non-zero exit code: 1
Error: Process completed with exit code 1.
```

**Comment by sffc** [comment]
Now that the P0 issue has been fixed (by #529), should we use this PR to pin the version of Rust in both nightly and stable CI builds?  And then hopefully unblock adding back CI caching?

**Comment by gregtatum** [comment]
I'm going to go ahead and mark this as ready for review, as the only outstanding CI failure is from:

PR #518
https://github.com/unicode-org/icu4x/runs/2040894278

**Review by Manishearth — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
RSLGTM for the nightly toolchain version.  As discussed, we should follow up with a PR to pin the stable version as well.

**Pull request merged**